src/azul/plugins/metadata/hca/transform.py

```diff
@@ -82,6 +86,123 @@
 pass_thru_uuid4: PassThrough[api.UUID4] = PassThrough()
 
 
+class OrganismAge(FieldType[JSON, str]):
+
+    def to_index(self, value: Optional[JSON]) -> str:
+        """
+        >>> a = OrganismAge()
+        >>> a.to_index({'value': '20', 'unit': 'year'})
+        '20 year'
+
+        >>> a = OrganismAge()
+        >>> a.to_index({'value': '20', 'unit': None})
+        '20'
+
+        >>> a = OrganismAge()
+        >>> a.to_index(None)
+        '~null'
+
+        >>> a = OrganismAge()
+        >>> a.to_index({'notval': '20', 'notu': 'year'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: A dictionary with entries for `value` and `unit` is required
+
+        >>> a = OrganismAge()
+        >>> a.to_index({'value': '', 'unit': 'year'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `value` entry must not be empty
+
+        >>> a = OrganismAge()
+        >>> a.to_index({'value': '20', 'unit': ''})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `unit` entry must not be empty
+
+        >>> a = OrganismAge()
+        >>> a.to_index({'value': None, 'unit': 'years'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: A `unit` entry must be accompanied by a `value` entry
+
+        >>> a = OrganismAge()
+        >>> a.to_index({'value': '20 ', 'unit': 'years'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `value` entry must not contain space characters
+
+        >>> a = OrganismAge()
+        >>> a.to_index({'value': 20, 'unit': 'years'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `value` entry must be of string type
+
+        >>> a = OrganismAge()
+        >>> a.to_index({'value': '20', 'unit': 'years '})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `unit` entry must not contain space characters
+
+        >>> a = OrganismAge()
+        >>> a.to_index({'value': '20', 'unit': True})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `unit` entry must be of string type
+
+        """
+        if value is None:
+            value, unit = [None, None]
```

Comment:
```suggestion
            value, unit = None, None
```

Reply:
@hannes-ucsc I took this section as the [ambiguity you described](https://github.com/DataBiosphere/azul/pull/1909#discussion_r489724920). When `None` is passed as a param to this method, it will return the expected index translation.